tests/integration/test_core.py

```diff
@@ -53,6 +55,33 @@ async def test_hello_kubecon_cache_header(ingress_ip: str):
     assert "content-cache-k8s" in response.headers["X-Cache-Status"]
 
 
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_service_reachable(service_ip: str):
+    """
+    arrange: given charm has been built, deployed and related to a dependent application
+    act: when the dependent application is queried via the service
+    assert: then the response is HTTP 200 OK.
+    """
+    response = requests.get(f"http://{service_ip}")
```

Comment:
As @weiiwang01 warned me before: "Add a timeout to requests.get, there's no default timeout for requests so it will wait indefinitely if there's no response." :-)